test/acceptance/test/ui_tests.go

```diff
@@ -75,16 +80,142 @@ var _ = Describe("Weave GitOps UI Test", func() {
 		Expect(webDriver.Destroy()).To(Succeed())
 	})
 
-	It("SmokeTest - Verify gitops can run UI without apps installed", func() {
+	It("SmokeTest - Verify gitops can add apps from the UI to an empty cluster", func() {
+		var addAppPage *pages.AddAppPageElements
+		var repoAbsolutePath string
+		tip := generateTestInputs()
+		appName := tip.appRepoName
+		private := true
+		appRepoRemoteURL := "ssh://git@github.com/" + GITHUB_ORG + "/" + tip.appRepoName + ".git"
 
-		dashboardPage := pages.Dashboard(webDriver)
-		expectedTitle := "Weave GitOps"
+		dashboardPage = pages.GetDashboardPageElements(webDriver)
+		addAppPage = pages.GetAddAppPageElements(webDriver)
 
-		By("Then I should be able to navigate to WeGO dashboard", func() {
-			Expect(webDriver.Navigate(WEGO_UI_URL)).To(Succeed())
-			str, _ := webDriver.Title()
-			Eventually(str, THIRTY_SECOND_TIMEOUT).Should(ContainSubstring(expectedTitle))
-			Expect(dashboardPage.ApplicationTab).Should(BeFound())
+		defer deleteRepo(tip.appRepoName, gitproviders.GitProviderGitHub, GITHUB_ORG)
+		defer deleteWorkload(tip.workloadName, tip.workloadNamespace)
+
+		By("When I navigate to Add Application page", func() {
+			Expect(dashboardPage.AddAppButton.Click()).To(Succeed())
+		})
+
+		By("Then I should see Add Applcation page", func() {
+			Expect(addAppPage.AddAppHeader.Text()).Should(ContainSubstring(addApplicationPageHeader))
 		})
+
+		By("When I create an app repo with workload that does not already exist", func() {
+			deleteRepo(tip.appRepoName, gitproviders.GitProviderGitHub, GITHUB_ORG)
+			repoAbsolutePath = initAndCreateEmptyRepo(tip.appRepoName, gitproviders.GitProviderGitHub, private, GITHUB_ORG)
+			gitAddCommitPush(repoAbsolutePath, tip.appManifestFilePath)
+		})
+
+		By("And I install gitops to my active cluster", func() {
+			installAndVerifyWego(WEGO_DEFAULT_NAMESPACE, appRepoRemoteURL)
+		})
+
+		By("And I add application details in Application form", func() {
+			_ = addAppPage.AppName.Fill(appName)
+			_ = addAppPage.AppRepoURL.Fill(appRepoRemoteURL)
+			_ = addAppPage.PathToManifests.Fill("/")
+		})
+
+		By("And default form values are present", func() {
+			Expect(addAppPage.AppNamespace.Attribute("value")).Should(ContainSubstring(WEGO_DEFAULT_NAMESPACE))
+			Expect(addAppPage.Branch.Attribute("value")).Should(ContainSubstring("main"))
+		})
+
+		By("And auto-merge is turned on", func() {
+			_ = addAppPage.AutoMergeCheck.Check()
+		})
+
+		By("And I submit the Add App form", func() {
+			Expect(addAppPage.SubmitButton.Click()).To(Succeed())
```

Comment:
If this doesn't check for the presence of anything on the cluster or that a git repo was populated, it isn't doing a full end-to-end-test.

Reply:
I was going to open up another ticket/PR for this and other changes like verifying `Application Details` page structure.. Its easier to add to the tests in increments - takes a long time to locate some of the elements and have them run successfully.

Reply:
I believe the github authentication will also be required for full app add test scenarios.